src/Bundle/Provider/ServiceGridProvider.php

```diff
@@ -15,35 +15,23 @@
 
 use Sylius\Bundle\GridBundle\Grid\GridInterface;
 use Sylius\Bundle\GridBundle\Registry\GridRegistryInterface;
-use Sylius\Component\Grid\Configuration\GridConfigurationExtenderInterface;
-use Sylius\Component\Grid\Configuration\GridConfigurationRemovalsHandler;
-use Sylius\Component\Grid\Configuration\GridConfigurationRemovalsHandlerInterface;
-use Sylius\Component\Grid\Definition\ArrayToDefinitionConverterInterface;
 use Sylius\Component\Grid\Definition\Grid;
 use Sylius\Component\Grid\Exception\UndefinedGridException;
+use Sylius\Component\Grid\Factory\GridFactoryInterface;
 use Sylius\Component\Grid\Provider\GridProviderInterface;
-use Webmozart\Assert\Assert;
 
 final class ServiceGridProvider implements GridProviderInterface
 {
-    private ArrayToDefinitionConverterInterface $converter;
-
     private GridRegistryInterface $gridRegistry;
 
-    private GridConfigurationExtenderInterface $gridConfigurationExtender;
-
-    private GridConfigurationRemovalsHandlerInterface $gridConfigurationRemovalsHandler;
+    private GridFactoryInterface $gridFactory;
 
     public function __construct(
-        ArrayToDefinitionConverterInterface $converter,
```

Comment:
I'm wondering about bc break. 
This is a final class, so maybe this is ok. @GSadee ?

Reply:
I'm afraid we have to handle this with these dependencies and the previous logic